activerecord/lib/active_record/connection_adapters.rb

```diff
@@ -26,6 +26,10 @@ module ConnectionAdapters
       autoload :ReferenceDefinition
     end
 
+    autoload_at "active_record/connection_adapters/postgresql/schema_definitions" do
+      autoload :ExclusionConstraintDefinition
```

Comment:
This autoload should not be necessary because it is explicitly required at https://github.com/rails/rails/blob/d921f203c356a0d6902a3f75067f08993f4e33fa/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb#L16